processor/src/logic/event-dispatcher.ts

```diff
@@ -101,7 +108,10 @@ export function createEventDispatcher({
           return handleEvent(observer, event, userProgress)
         })
 
-      const badgesToGrant = (await Promise.all(asyncResults)).filter(Boolean).flat()
+      const results = await Promise.all(asyncResults)
+
+      const badgesToGrant = results.filter(({ ok, result }) => ok && !!result).map(({ result }) => result)
+      const handlersToRetry = results.filter(({ ok }) => !ok)
```

Comment:
I would keep it for now but add an underscore so we by-pass the lint warning

```suggestion
      const _handlersToRetry = results.filter(({ ok }) => !ok)
```